node/v2/handler.js

```diff
@@ -495,10 +495,7 @@ TChannelV2Handler.prototype.handleError = function handleError(errFrame, callbac
     var id = errFrame.id;
     var code = errFrame.body.code;
     var message = String(errFrame.body.message);
-    var err = v2.ErrorResponse.CodeErrors[code]({
-        originalId: id,
-        message: message
-    });
+    var err = new v2.ErrorResponse.CodeErrors[code](id, message);
```

Comment:
Someone has to create an error object somewhere. 

We cannot emit an errorEvent without an error object.

Let's emit an errorFrameEvent and have the endpoint handler or client or non-relay somewhere convert an error frame into an error object right before we call the callback as part of the public interface.